library.c

```diff
@@ -359,7 +359,8 @@ redis_check_eof(RedisSock *redis_sock, zend_bool no_retry, zend_bool no_throw)
         for (retry_index = 0; !no_retry && retry_index < redis_sock->max_retries; ++retry_index) {
             /* close existing stream before reconnecting */
             if (redis_sock->stream) {
-                redis_sock_disconnect(redis_sock, 1);
+                /* reconnect no need to reset mode, it will cause pipeline mode socket exception */
```

Comment:
If we're not atomic here, we may need to just throw an exception.

We could be halfway through a `MULTI` transaction or in the middle of reading commands delivered via pipeline.  PhpRedis could get out of sync in that case.

Reply:
i think it's okay for this case.
for `MULTI` transaction, it will not reconnect ([see this line](https://github.com/phpredis/phpredis/blob/develop/library.c#L354)). Therefore, it will not reach the line and throw an exception.

Reply:
I'm sorry, you are correct.

In that case, do we need an argument for whether or not we reset the mode?  If we disconnect shouldn't we just always reset it to ATOMIC?